### PR TITLE
Feature/server connection processing

### PR DESCRIPTION
**Pull request by Mehanari**
Added logic for server connection process and visuals to represent it.

### CODE REVIEW

File: Assets/Src/Components/TextSwitchingAnimation.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    /// </summary>
+    public class TextSwitchingAnimation : MonoBehaviour
+    {
+       [SerializeField] private TextMeshProUGUI textMesh;
+       [SerializeField] private StringsListConfig stringsListConfig;
+       [SerializeField] private FloatValueConfig intervalConfig;
+       private float _elapsedTime = 0;
+       private int _currentStringIndex = 0;
+       
+       private void Update()
```
Comment (GGLinker) [inline Assets/Src/Components/TextSwitchingAnimation.cs:19]: Do we rly need timer implementation via Update()? It'll be ticking besides of the fact whether parent GameObject is active or not. We'll need to Destroy() parent GO to force stop such an update. I'd prefer to refactor this with usage of async awaits chain or some kind of Coroutine

File: Assets/Src/General/FloatValueConfig.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using UnityEngine;
+
+namespace Src.General
```
Comment (GGLinker) [inline Assets/Src/General/FloatValueConfig.cs:3]: Seems a _lot_ overcomplicated to me. Mb this should be a simple SF in a corresponding classes, or if such a config class has been created for low-range purpose bunch of another classes, it'd be better to move such a SF into those parent interfaces' declarations?

File: Assets/Src/General/StringsListConfig.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+
+namespace Src.General
+{
+    [CreateAssetMenu(menuName = "Configs/StringsListConfig", fileName = "StringsListConfig", order = 0)]
+    public class StringsListConfig : ScriptableObject
```
Comment (GGLinker) [inline Assets/Src/General/StringsListConfig.cs:7]: More understandable move with such a config, but still it's too low-level structure to extract while we have such a powerful tool as SerializeField attribute

File: Assets/Src/NetworkingModule/ClientWrapper.cs [hunk 4]
```diff
@@ -1,19 +1,58 @@
...
-    public class ClientWrapper : IMessageSender
+    public class ClientWrapper : IClientWrapper
     {
         public Client Client { get; private set; }
+        
+        public event EventHandler<DisconnectedEventArgs> Disconnected;
+        
+        public event EventHandler Connected;
+        
+        public event EventHandler<ConnectionFailedEventArgs> ConnectionFailed;
```
Comment (GGLinker) [inline Assets/Src/NetworkingModule/ClientWrapper.cs:14]: If this class would be used not only within some non-large family of classes / namespace, mb it'd be better to make these events of **Action** type?

File: Assets/Src/NetworkingModule/ClientWrapper.cs [hunk 5]
```diff
@@ -1,19 +1,58 @@
...
+        }
+        
+        private void OnConnected(object sender, EventArgs e)
+        {
+            Connected?.Invoke(this, e);
+        }
+        
+        private void OnConnectionFailed(object sender, ConnectionFailedEventArgs e)
+        {
+            ConnectionFailed?.Invoke(this, e);
```
Comment (GGLinker) [inline Assets/Src/NetworkingModule/ClientWrapper.cs:48]: Despite of the fact that EventHandler is stated in C# standard as recommended solution, such meaningless mentions of <_this, e_> looks a bit weird

File: Assets/Src/NetworkingModule/IClientWrapper.cs [hunk 6]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using Riptide;
+
+namespace Src.NetworkingModule
+{
+    public interface IClientWrapper : IMessageSender
+    {
+        bool Connect(string hostAddress, int maxConnectionAttempts = 5, byte messageHandlerGroupId = 0,
+            Message message = null);
```
Comment (GGLinker) [inline Assets/Src/NetworkingModule/IClientWrapper.cs:9]: I think it's not a good idea to make default method parameter values in interface bc it feels like side effect. And besides of that, such a magic numbers organically mix up into smth like config which we use anywhere we can. Why not there too?

File: Assets/Src/GameplayPresenter/ServerConnection/ServerConnectionPresenter.cs [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        {
+            _view.HideConnectingMessage();
+        }
+
+
+        public void ConnectToServer()
+        {
+            _view.ShowConnectingMessage();
+            var hostAddress = _connectionConfig.HostAddress;
+            var maxConnectionAttempts = _connectionConfig.MaxConnectionAttempts;
```
Comment (GGLinker) [inline Assets/Src/GameplayPresenter/ServerConnection/ServerConnectionPresenter.cs:39]: In IMessageSender (and in IClientWrapper) we use method parameter to get max connection attempts count. I think we should apply such a solution there too.

**Review by GGLinker — CHANGES_REQUESTED** [review]

File: Assets/Src/Components/TextSwitchingAnimation.cs (hunk 1 above)
Comment (Mehanari) [reply Assets/Src/Components/TextSwitchingAnimation.cs:19]: Update automatically stops if parent game object is deactivated.

**Review by GGLinker — APPROVED** [review]

**Pull request merged**